### PR TITLE
Add Jordan holidays

### PR DESCRIPTION
**Pull request by firaskafri**
## Proposed change

Added Jordan (Country) to supported countries.

## Type of change

- [X] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [X] I've followed the [contributing guidelines][contributing-guidelines]
- [X] I've run `make pre-commit`, it didn't generate any changes
- [X] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/dev/docs/source

### CODE REVIEW

File: holidays/countries/jordan.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars.gregorian import GREGORIAN_CALENDAR
+from holidays.groups import InternationalHolidays, IslamicHolidays, ChristianHolidays
+from holidays.holiday_base import HolidayBase
+
+
+class Jordan(HolidayBase, IslamicHolidays, ChristianHolidays, InternationalHolidays):
```
Comment (arkid15r) [inline holidays/countries/jordan.py:20]: Use alphabetical order after the base class:
```suggestion
class Jordan(HolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
```

File: holidays/countries/jordan.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("رأس السنة الميلادية"))
+
+        # Independance Day.
+        self._add_holiday_may_25(tr("عيد الاستقلال"))
+
+        # Labor Day.
+        self._add_labor_day(tr("عيد العمال"))
```
Comment (arkid15r) [inline holidays/countries/jordan.py:47]:
```suggestion
        # Labor Day.
        self._add_labor_day(tr("عيد العمال"))

        # Independance Day.
        self._add_holiday_may_25(tr("عيد الاستقلال"))

```

File: holidays/locale/ar/LC_MESSAGES/JO.po [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/python-holidays
+#  License: MIT (see LICENSE file)
+#
+# Jordan holidays.
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Python Holidays 0.44\n"
```
Comment (arkid15r) [inline holidays/locale/ar/LC_MESSAGES/JO.po:17]:
```suggestion
"Project-Id-Version: Python Holidays 0.47\n"
```

File: holidays/locale/en_US/LC_MESSAGES/JO.po [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/python-holidays
+#  License: MIT (see LICENSE file)
+#
+# Jordan holidays en_US localization.
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Python Holidays 0.44\n"
```
Comment (arkid15r) [inline holidays/locale/en_US/LC_MESSAGES/JO.po:17]:
```suggestion
"Project-Id-Version: Python Holidays 0.47\n"
```

File: holidays/registry.py [hunk 5]
```diff
@@ -158,7 +159,12 @@
         "UMI",
         "HolidaysUM",
     ),
-    "united_states_virgin_islands": ("UnitedStatesVirginIslands", "VI", "VIR", "HolidaysVI"),
+    "united_states_virgin_islands": (
+        "UnitedStatesVirginIslands",
+        "VI",
+        "VIR",
+        "HolidaysVI",
+    ),
```
Comment (arkid15r) [inline holidays/registry.py:167]: This doesn't look related, could you revert it?

File: holidays/countries/jordan.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/python-holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars.gregorian import GREGORIAN_CALENDAR
+from holidays.groups import InternationalHolidays, IslamicHolidays, ChristianHolidays
```
Comment (arkid15r) [inline holidays/countries/jordan.py:16]: Alphabetical order for the imports:
```suggestion
from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
```

**Review by arkid15r — COMMENTED** [review]
Hey @firaskafri,
thanks for adding another country support!

Please see some minor comments to address before merging this PR:

**Comment by arkid15r** [comment]
@abh31000, could you review this when you get a chance?

Thank you!

File: holidays/countries/jordan.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(self)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("رأس السنة الميلادية"))
+
+        # Independance Day.
+        self._add_holiday_may_25(tr("عيد الاستقلال"))
```
Comment (abh31000) [inline holidays/countries/jordan.py:44]: This is more correct. It's unified across all the other AR locales
```suggestion
        self._add_holiday_may_25(tr("عيد الإستقلال"))
```

File: holidays/locale/en_US/LC_MESSAGES/JO.po [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+#. Eid al-Fitr Holiday.
+msgid "عطلة عيد الفطر"
+msgstr "Eid al-Fitr Holiday"
+
+#. Labor Day.
+msgid "عيد العمال"
+msgstr "Labor Day"
+
+#. Independance Day.
+msgid "عيد الاستقلال"
```
Comment (abh31000) [inline holidays/locale/en_US/LC_MESSAGES/JO.po:54]:
```suggestion
msgid "عيد الإستقلال"
```

File: tests/countries/test_jordan.py [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    def test_2024(self):
+        self.assertHolidays(
+            Jordan(years=2024),
+            ("2024-01-01", "رأس السنة الميلادية"),
+            ("2024-02-08", "(تقدير) ليلة المعراج"),
+            ("2024-04-10", "(تقدير) عيد الفطر"),
+            ("2024-04-11", "(تقدير) عطلة عيد الفطر"),
+            ("2024-04-12", "(تقدير) عطلة عيد الفطر"),
+            ("2024-05-01", "عيد العمال"),
+            ("2024-05-25", "عيد الاستقلال"),
```
Comment (abh31000) [inline tests/countries/test_jordan.py:36]:
```suggestion
            ("2024-05-25", "عيد الإستقلال"),
```

**Review by abh31000 — COMMENTED** [review]
This is more correct. And it's unified across all the other AR locales

**Comment by firaskafri** [comment]
Hey @arkid15r @abh31000 !

Thank you for the review, all addressed now.

File: holidays/countries/jordan.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    """
+
+    country = "JO"
+    default_language = "ar"
+    # %s (estimated).
+    estimated_label = tr("(تقدير) %s")
+    supported_languages = ("ar", "en_US")
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self, GREGORIAN_CALENDAR)
```
Comment (KJhellico) [inline holidays/countries/jordan.py:34]:
```suggestion
        ChristianHolidays.__init__(self)
```
It's default value.

File: holidays/countries/jordan.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+#
+#  Authors: Vacanza Team and individual contributors (see AUTHORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/python-holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars.gregorian import GREGORIAN_CALENDAR
```
Comment (KJhellico) [inline holidays/countries/jordan.py:15]:
```suggestion
```

File: holidays/countries/jordan.py [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("رأس السنة الميلادية"))
+
+        # Labor Day.
+        self._add_labor_day(tr("عيد العمال"))
+
+        # Independance Day.
```
Comment (KJhellico) [inline holidays/countries/jordan.py:46]:
```suggestion
        # Independence Day.
```

File: holidays/locale/ar/LC_MESSAGES/JO.po [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+
+#. Eid al-Fitr Holiday.
+msgid "عطلة عيد الفطر"
+msgstr ""
+
+#. Labor Day.
+msgid "عيد العمال"
+msgstr ""
+
+#. Independance Day.
```
Comment (KJhellico) [inline holidays/locale/ar/LC_MESSAGES/JO.po:53]:
```suggestion
#. Independence Day.
```

File: holidays/locale/en_US/LC_MESSAGES/JO.po [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+msgid "عطلة عيد الفطر"
+msgstr "Eid al-Fitr Holiday"
+
+#. Labor Day.
+msgid "عيد العمال"
+msgstr "Labor Day"
+
+#. Independance Day.
+msgid "عيد الإستقلال"
+msgstr "Independance Day"
```
Comment (KJhellico) [inline holidays/locale/en_US/LC_MESSAGES/JO.po:55]:
```suggestion
#. Independence Day.
msgid "عيد الإستقلال"
msgstr "Independence Day"
```

File: tests/countries/test_jordan.py [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+    def test_l10n_en_us(self):
+        self.assertLocalizedHolidays(
+            "en_US",
+            ("2024-01-01", "New Year's Day"),
+            ("2024-02-08", "Isra and Miraj (estimated)"),
+            ("2024-04-10", "Eid al-Fitr (estimated)"),
+            ("2024-04-11", "Eid al-Fitr Holiday (estimated)"),
+            ("2024-04-12", "Eid al-Fitr Holiday (estimated)"),
+            ("2024-05-01", "Labor Day"),
+            ("2024-05-25", "Independance Day"),
```
Comment (KJhellico) [inline tests/countries/test_jordan.py:73]:
```suggestion
            ("2024-05-25", "Independence Day"),
```

**Review by KJhellico — COMMENTED** [review]
@firaskafri, thank you for your contribution! A couple of fixes:

**Review by arkid15r — COMMENTED** [review]
Hey @firaskafri. Thanks for fixing all the comments! The PR looks very decent, I'm going to take another look today and merge it.

Here is an optional comment to address -- we've introduced AUTHORS file quite recently. Would you like to update it and add your name to the list?


@abh31000 thanks for reviewing the holiday names!

**Review by arkid15r — COMMENTED** [review]
Hi @firaskafri, I've updated the `AUTHORS` file and merging this PR in.

Thanks a lot for adding another country holidays support!

**Review by arkid15r — APPROVED** [review]
